packages/lib/src/classes/spot.ts

```diff
@@ -36,87 +36,59 @@ export class Spot {
                 console.debug(
                     `Received ${beamTaskResponse.status} with message ${error}`,
                 );
                 throw new Error(`Unable to create new beam task.`);
             }
-            this.currentTask = (await beamTaskResponse.json()).id;
-
-            let responseCount: number = 0;
 
-            do {
-                const beamResponses: Response = await fetch(
-                    `${this.url}tasks/${this.currentTask}?wait_count=${responseCount + 1}`,
-                    {
-                        credentials: import.meta.env.PROD ? "include" : "omit",
-                        signal: controller?.signal,
-                    },
-                );
-
-                if (!beamResponses.ok) {
-                    const error: string = await beamResponses.text();
-                    console.debug(
-                        `Received ${beamResponses.status} with message ${error}`,
-                    );
-                    throw new Error(
-                        `Error then retrieving responses from Beam. Abborting requests ...`,
-                    );
-                }
+            console.log(`Created new Beam Task with id ${this.currentTask}`)
 
-                const beamResponseData: Array<BeamResult> =
-                    await beamResponses.json();
+            let eventSource = new EventSource(`${this.url.toString()}beam/${this.currentTask}?wait_count=${this.sites.length}`)
+            eventSource.addEventListener("new_result", (message) => {
+                const response: BeamResult = JSON.parse(message.data)
+                if (response.task !== this.currentTask) return
+                let site: string = response.from.split(".")[1]
+                let status: Status = response.status
+                let body: SiteData = (status === "succeeded") ? JSON.parse(atob(response.body)) : null;
 
-                const changes = new Map<string, Site>();
-                beamResponseData.forEach((response: BeamResult) => {
-                    if (response.task !== this.currentTask) return;
-                    const site: string = response.from.split(".")[1];
-                    const status: Status = response.status;
-                    const body: SiteData =
-                        status === "succeeded"
-                            ? JSON.parse(atob(response.body))
-                            : null;
+                responseStore.update((store: ResponseStore): ResponseStore => {
+                    store.set(site, { status: status, data: body })
+                    return store;
+                })
+            })
 
-                    // if the site is already in the store and the status is claimed, don't update the store
-                    if (this.storeCache.get(site)?.status === status) return;
+            // read error events from beam
+            eventSource.addEventListener("error", (message) => {
+                console.log(`Beam returned error ${message}`)
```

Comment:
If beam encounters an error, maybe we should inform the user. I'm pretty sure we should do something like this, but not in this PR.

Reply:
Yes, i think that is a good idea. Maybe we should open an issue for this. I'think we should just provide an api for the applications to listen for such events, as each application using lens will maybe have their own way of handling it (e.g. showing the notification in the top of the application or as a box in the right bottom corner)